core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -364,4 +376,123 @@ private static void checkCatalogType(
           catalogIdent, catalog.type(), privilege);
     }
   }
+
+  public static List<String> getMetadataObjectLocation(
+      NameIdentifier ident, Entity.EntityType type) {
+    List<String> locations = new ArrayList<>();
+    MetadataObject metadataObject;
+    try {
+      metadataObject = NameIdentifierUtil.toMetadataObject(ident, type);
+    } catch (IllegalArgumentException e) {
+      LOG.warn("Illegal argument exception for metadata object %s type %s", ident, type, e);
+      return locations;
+    }
+
+    String metalake =
+        (type == Entity.EntityType.METALAKE ? ident.name() : ident.namespace().level(0));
+    try {
+      switch (metadataObject.type()) {
+        case METALAKE:
```

Comment:
Do we need metalake location? Which case?

Reply:
For example, When we grant `CreateSchema` privilege under a metalake, Just like my ITs `TestChainedAuthorizationIT`.
```
    SecurableObject securableObject =
        SecurableObjects.ofMetalake(
            metalakeName, Lists.newArrayList(Privileges.CreateSchema.allow()));
```

Reply:
Maybe catalog's locations aren't enough. We need know the schema location and table location.

Reply:
I will consider adding other locations in the next PR.
Currently, We only need catalog location.